cmd/crank/push.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"os"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"github.com/spf13/afero"
+
+	"github.com/crossplane/crossplane/pkg/xpkg"
+)
+
+// PushCmd pushes a package.
+type pushCmd struct {
+	Configuration pushConfigCmd   `cmd:"" help:"Push a Configuration package."`
+	Provider      pushProviderCmd `cmd:"" help:"Push a Provider package."`
+
+	Package string `short:"f" help:"Path to package. If not specified and only one package exists in current directory it will be used."`
+}
+
+// Run runs the push cmd.
+func (c *pushCmd) Run(child *childArg) error {
```

Comment:
Logic in the run function of a command with child node subcommands will be invoked after the subcommands are. This means that we can use common functionality in the parents and just use the child leaf nodes for value setting, etc.